rust/src/detect/vlan_id.rs

```diff
@@ -0,0 +1,177 @@
+/* Copyright (C) 2024 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+use super::uint::{detect_parse_uint, DetectUintData};
+use std::ffi::CStr;
+use std::str::FromStr;
+
+#[no_mangle]
+pub static ANY_VLAN_LAYER: i8 = i8::MIN;
+
+#[no_mangle]
+pub static ALL_VLAN_LAYERS: i8 = i8::MAX;
```

Comment:
These constants could be prefixed with `DETECT_VLAN_ID_`

Reply:
And should be `pub const` instead of `pub static` to be picked up by cbindgen like `STREAM_START` is done